Sword attack - Cliff

TwoSteps555 opened the pull request:
Press F to attack! Sword animation is working. When the player attacks, an "attack volume" is enabled briefly in front of the player. There aren't any targets to attack yet, but everything is set up for if/when we want to add some stuff to attack.

TwoSteps555 committed:
Added sword attack logic

There's no animation attached to the sword attack, also there's nothing to "attack" yet so attacking doesn't... really... do anything...

TwoSteps555 committed:
Added "Attacking" state to scripts

Animation still isn't happening, gonna ask around

TwoSteps555 committed:
Animation is working!

Animation won't play if player is jumping, but that will likely require a "jump attack" animation anyway so I'm calling this done for now.

MinerSheep committed:
animation doesnt work, but this is fine

MinerSheep committed:
Merge branch 'development-donotdelete' into SwordAttack-Cliff

MinerSheep reviewed (changes requested):
Animation needs fixing.  Vomunar suggested "create a new state
like a new default state
cus atttacking is not a standalone state"

TwoSteps555 commented:
>animation doesnt work, but this is fine

Could you elaborate on "doesn't work"? I just tested it, and the animation plays when the player presses the attack button.
The only time the animation doesn't play is if the player is jumping, but I left that out intentionally in case we want to add a "jumping attack" animation.

MinerSheep commented:
@TwoSteps555  That is very strange.  No, I can't get it to activate at all, jumping or not jumping...  

I'm looking into why this might be happening.  Perhaps you could try making a duplicate animation controller and attaching the duplicate to the player?

The attack hitbox is working fine though

TwoSteps555 committed:
Updated sword_man.controller

Sword attack tested and working, hopefully this change is all the main branch needed.
